src/models/data.cpp

```diff
@@ -133,6 +133,158 @@ ItemCost buildCallerCalleeResult(const BottomUp& data, const Costs& bottomUpCost
+        *size = prefix.size();
+        return true;
+    }
+
+    return false;
+}
+
+int findSameDepth(const QStringRef& str, int offset, QChar ch, bool returnNext = false)
+{
+    const int size = str.size();
+
+    int depth = 0;
+    for (; offset < size; ++offset) {
+        const auto current = str[offset];
+        if (current == QLatin1Char('<') || current == QLatin1Char('(')) {
+            ++depth;
+        }
+        else if (current == QLatin1Char('>') || current == QLatin1Char(')')) {
+            --depth;
+        }
+
+        if (depth == 0 && current == ch) {
+            return offset + (returnNext ? 1 : 0);
+        }
+    }
+    return -1;
+}
+
+QString prettifySymbol(const QStringRef& str)
+{
+    int pos = 0;
+    do {
+        pos = str.indexOf(QLatin1String("std::"), pos);
+        if (pos == -1) {
+            return str.toString();
+        }
+        pos += 5;
+        if (pos == 5
+            || str[pos - 6] == QLatin1Char('<')
+            || str[pos - 6] == QLatin1Char(' ')
+            || str[pos - 6] == QLatin1Char('(')) {
+            break;
+        }
+    } while (true);
+
+    auto result = str.left(pos).toString();
+    auto symbol = str.mid(pos);
+
+    int end;
+    if (startsWith(symbol, QLatin1String("__cxx11::"), &end)
+        || startsWith(symbol, QLatin1String("__1::"), &end)) {
+        // Strip libstdc++/libc++ internal namespace
+        symbol = symbol.mid(end);
+    }
+
+    if (startsWith(symbol, QLatin1String("basic_string<"), &end)) {
+        const int comma = findSameDepth(symbol, end, QLatin1Char(','));
+        if (comma != -1) {
+            const auto type = symbol.mid(end, comma - end);
+            if (type == QLatin1String("char")) {
+                result += QLatin1String("string");
+            }
+            else if (type == QLatin1String("wchar_t")) {
+                result += QLatin1String("wstring");
+            }
+            else {
+                result += symbol.left(end);
+                result += type;
+                result += QLatin1Char('>');
+            }
+            end = findSameDepth(symbol, 0, QLatin1Char('>'), true);
+            symbol = symbol.mid(end);
+
+            if (startsWith(symbol, QLatin1String("::basic_string("), &end)
+                || startsWith(symbol, QLatin1String("::~basic_string("), &end)) {
+                result += QLatin1String("::");
+                if (symbol[2] == QLatin1Char('~')) {
+                    result += QLatin1Char('~');
+                }
+                if (type == QLatin1String("wchar_t")) {
+                    result += QLatin1Char('w');
+                }
+                else if (type != QLatin1String("char")) {
+                    result += QLatin1String("basic_");
+                }
+                result += QLatin1String("string(");
+                symbol = symbol.mid(end);
+            }
+        }
+    }
+    else if (startsWith(symbol, QLatin1String("vector<"), &end)
```

Comment:
please introduce a lambda that's used here instead of the long conditional chain, and within that lambda use an initializer list over some type names, e.g. something like:

```
const auto types = {QLatin1String("set"), ...};
for (const auto &type : types) {
    if (startsWith(symbol, type, &end))
        return true;
}